### PR TITLE
explicitly use the utf8 encoding to read and write files

### PR DESCRIPTION
**Pull request by yfpeng**
Fix the UnicodeEncodeError while building the source on Center OS.

### CODE REVIEW

**Comment by agitter** [comment]
Can you please remove the modifications to `tags.tsv` and `03_categorize.md` so we can review those separately in #567?  I saw that pull request but will need more time to look it over.

Did the build error occur within the `deepreview` conda environment or a different environment?

@dhimmel these changes to `references.py` and `citations.py` look okay to me.

**Comment by yfpeng** [comment]
> Can you please remove the modifications to tags.tsv and 03_categorize.md so we can review those separately in #567? I saw that pull request but will need more time to look it over.

Fixed.

> Did the build error occur within the deepreview conda environment or a different environment?

It was within the deepreview conda.

File: build/citations.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from hashlib import blake2b
 import pathlib
 import re
+import sys
```
Comment (dhimmel) [inline build/citations.py]: Unused import?

Comment (yfpeng) [inline build/citations.py]: > Unused import?
Removed

> What is Center OS? http://www.centeros.com/ or https://www.centos.org/ ?
 Sorry. It is CentOS.

> Instead it defaults to whatever `locale.getpreferredencoding()` returns.

On the system we are using, it returns "ANSI_X3.4-1968"

**Pull request merged**